File: src/test/java/org/example/testing/RepeatRemoverTest.java
package org.example.testing;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class RepeatRemoverTest {

  private RepeatRemover subj = new RepeatRemover();

  @Test
  void remove() {
    Assertions.assertEquals("ABC", subj.remove("AABCC"));
  }

  @Test
  void staticRemove() {
    Assertions.assertEquals("ABC", RepeatRemover.staticRemove("AABCC"));
  }
}

File: src/test/clojure/repeat_remover_test.clj
(ns repeat-remover-test
    (:require [clojure.test :refer [deftest is]])
  (:import (org.example.testing RepeatRemover)
           (org.junit.jupiter.api Assertions))
    (:gen-class
      :name org.example.testing.RepeatRemoverClojureTest
      :main false
      :methods [[^{org.junit.jupiter.api.Test {}}
                 testStaticCall [] void]
                [^{org.junit.jupiter.api.Test {}}
                 testInstanceCall [] void]]
      :prefix "")
  )

(def subj (RepeatRemover.))

(defn testStaticCall [this]
  (Assertions/assertEquals "ABC" (RepeatRemover/staticRemove "AABCC")))

(defn testInstanceCall [this]
  (Assertions/assertEquals "ABC" (.remove subj "AABCC")))
